### DISCUSSION TITLE
Failing to compile on Pi4/Buster

### DISCUSSION THREAD

**Discussion by Exarkuniv** [diasurgical/DevilutionX#1249] [Q&A] [answered]
trying to compile devilutionX on my Raspberry Pi4
running Buster/RetroPie

i download the lastet source 
i use 
 cd build
    cmake ..
    make -j3
and i get this

[ 18%] Building CXX object CMakeFiles/devilution.dir/Source/objdat.cpp.o
/home/pi/RetroPie-Setup/tmp/build/devilutionx/Source/multi.cpp: In function ‘dvl::BOOL dvl::NetInit(dvl::BOOL, dvl::BOOL*)’:
/home/pi/RetroPie-Setup/tmp/build/devilutionx/Source/multi.cpp:741:54: error: expected primary-expression before ‘;’ token
   sgGameInitInfo.versionMajor = PROJECT_VERSION_MAJOR;
                                                      ^
/home/pi/RetroPie-Setup/tmp/build/devilutionx/Source/multi.cpp:742:54: error: expected primary-expression before ‘;’ token
   sgGameInitInfo.versionMinor = PROJECT_VERSION_MINOR;
                                                      ^
/home/pi/RetroPie-Setup/tmp/build/devilutionx/Source/multi.cpp:743:54: error: expected primary-expression before ‘;’ token
   sgGameInitInfo.versionPatch = PROJECT_VERSION_PATCH;
                                                      ^
make[2]: *** [CMakeFiles/devilution.dir/build.make:570: CMakeFiles/devilution.dir/Source/multi.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:147: CMakeFiles/devilution.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
/home/pi/RetroPie-Setup

not sure what it is telling me Thanks for any pointers

**Comment by AJenbo** [comment, marked as answer]
> i download the lastet source

Use Git or provide the version manually to CMake.

**Reply by Exarkuniv** [reply]
sorry for my naiveness how to i provide the version to cmake

**Reply by Exarkuniv** [reply]
i was able to follow your guild on how to install it on the retropie, and it works perfect. 

i was just trying to get the install script that someone made to compile it from source to work.
if i use the script. it fails

if i run the commands (which match what the script has) it works fine with out a issue. 
that very confusing, lol

**Reply by AJenbo** [reply]
I'll need to see the commands and or guide to be of any help with that :)

If you use CMake from a CLI this is how to provide the version manually:
```bash
cmake .. -DVERSION_NUM=1.1.0 -DVERSION_SUFFIX=TEST_VERSION
```

**Reply by Exarkuniv** [reply]
thank you for that, ill give that a shot. 
here is the script that we are using. i did not write it, but im just trying to fix it for others 

https://gist.github.com/Exarkuniv/a634dce5b7d7fb5c3c27dc5e02eec851

**Reply by AJenbo** [reply]
Not clear, maybe you just need to add git as a dependency. But I don't know if it fetches the repo or just a file dump. How are versions normally handled?

Another thing is that this is missing devilutionx.mpq

**Reply by Exarkuniv** [reply]
it seems that changing that line in the script got it to work perfectly 
thank you for all your help

**Reply by Exarkuniv** [reply]
> 
> 
> Not clear, maybe you just need to add git as a dependency. But I don't know if it fetches the repo or just a file dump. How are versions normally handled?
> 
> Another thing is that this is missing devilutionx.mpq

i will take a look into that and fix it

**Comment by Exarkuniv** [comment]
is the devilutionx.mpq the file that gets compiled. 
when i looked for it the only thing came up in the Packaging/resources

but once i put the game files in it works perfectly

**Reply by AJenbo** [reply]
No compiling produces the game application.

devilutionx.mpq gives better widescreen support and some other corrections, it will grow over time as we add more assets.

**Reply by Exarkuniv** [reply]
ahh ok. 
where should the devilutionx.mpq go. 
in with the game files or the devilutionx file

**Reply by AJenbo** [reply]
Same as other mpq files

**Reply by Exarkuniv** [reply]
thanks for all the help. it is working great.
